### ISSUE TITLE
Intersection semantics 

### ISSUE THREAD

**Issue by Rhuagh** [kvark/collision-rs#27]
We should define what a border hit should do for all intersection tests.

**Comment by Rhuagh** [comment]
My opinion is that border hits should not count, but that's just an opinion.

**Comment by kvark** [comment]
One action item to take - see what other libraries are doing.
In general, I'm fine with either decision as long as it's consistent ;)

**Comment by Rhuagh** [comment]
Yeah, I tried finding information on this, but my google skills apparently aren't up to the task, and I'm not sure what open libraries there are except for ncollide.

**Comment by Rhuagh** [comment]
ncollide seems to report a hit on border intersections. It's hard to tell ...

**Comment by msiglreith** [comment]
Related point would be checking the returned values on edge cases. ncollide for example had issues at 0 penetration depth, resulting in bogus normals returned from the hit test.
